src/router/storeSync/PositionUrlParamConfig.class.js

```diff
@@ -0,0 +1,229 @@
+        value === '' ? 0 : parseFloat(value)
+    )
+    store.dispatch('setCameraPosition', {
+        x,
+        y,
+        z,
+        yaw,
+        pitch,
+        roll,
+    })
+}
+
+function onlyShowIfNotZero(value) {
+    if (value === 0) {
+        return ''
+    }
+    return `${value}`
+}
+
+function generateCameraUrlParamFromStoreValues(store) {
+    if (store.state.ui.showIn3d) {
+        let { x, y, z, yaw, pitch, roll } = store.state.position.camera
+        return [x, y, z, yaw, pitch, roll].map((value) => onlyShowIfNotZero(value)).join(',')
+    }
+    return null
+}
+
+function readValueFromObjectOrReturnNull(object, paramName, type) {
+    if (object && paramName in object) {
+        return type(object[paramName])
+    }
+    return null
+}
+
+function parseCameraFromQuery(query) {
+    const camera = readValueFromObjectOrReturnNull(query, 'camera', String)
+    if (camera) {
+        let cameraValues = camera.split(',')
+        // the split must have 6 components (x, y, z, pitch, yaw and roll)
+        if (cameraValues.length === 6) {
+            // parsing to number all values (default to 0 if the value is empty)
+            cameraValues = cameraValues.map((value) => (value === '' ? 0 : Number(value)))
+            const [x, y, z, pitch, yaw, roll] = cameraValues
+            return {
+                x,
+                y,
+                z,
+                pitch,
+                yaw,
+                roll,
+            }
+        }
+    }
+    return null
+}
+
+function populateStoreWithParam(store, query, paramName, storeValue, dispatchTo) {
+    if (query && query.hasOwnProperty(paramName)) {
+        const value = Number(query[paramName])
+        if (value !== storeValue) {
+            return store.dispatch(dispatchTo, value)
+        }
+    }
+}
+
+/**
+ * Definition of a set of URL params to store the position of the map in the URL. These params will
+ * change depending on the mode of rendering of the map (2D or 3D)
+ *
+ * In 2D it will generate lon/lat/z params.
+ *
+ * In 3D it will generate a unique camera URL param that will be a concat of all relevant camera
+ * values (x,y,z,yaw,pitch,roll)
+ *
+ * This param config class takes for granted that the store's flag showIn3d has been processed
+ * before it is then called to do its job. Meaning that if the flag has changed in the URL, but the
+ * store is out of sync, things will break.
+ */
+export default class PositionUrlParamConfig extends AbstractParamConfig {
+    constructor() {
+        super(
+            'camera',
+            'setCameraPosition,setCenter,setZoom,setShowIn3d',
+            dispatchCameraPositionFromUrlIntoStore,
+            generateCameraUrlParamFromStoreValues,
+            false,
+            String
+        )
+    }
+
+    /**
+     * Even though it is not used below in populate functions, this is required so that the
+     * routerStoreSyncPlugin can work properly
```

Comment:
I don't understand the comment, if this is not used why not simply return null ?

Reply:
At first I removed it as I saw it wasn't used below, then I had issue at app startup where params were not parsed.
This was due to the fact that the router plugin uses `readQueryValue` to determine if a dispatch must occur (instead of using `valuesAreDifferentBetweenQueryAndStore(query, store)`, maybe a switch to this would be more logical?!)